addons/website_sale/tests/test_website_editor.py

```diff
@@ -116,79 +122,123 @@ def test_extra_images_with_new_variant(self):
+            i1, i2, i3, i4, i5, i6 = self._get_product_image_data()
             self.WebsiteSaleController.resequence_product_image(
                 images[2]._name,
                 images[2].id,
                 'left',
             )
+            self.env['product.image'].invalidate_model()
+            self.assertListEqual(self._get_product_image_data(), [i1, i3, i2, i4, i5, i6])
+
+    def test_resequence_image_right(self):
+        self._create_product_images()
+        with MockRequest(self.product.env, website=self.website):
             images = self.product._get_images()
-            self.assertEqual(images[2].image_1920, data_target)
-            self.assertEqual(images[1].image_1920, data_source)
-            # Test moving one to the right
-            data_source = images[2].image_1920
-            data_target = images[3].image_1920
+            i1, i2, i3, i4, i5, i6 = self._get_product_image_data()
             self.WebsiteSaleController.resequence_product_image(
                 images[2]._name,
                 images[2].id,
                 'right',
             )
+            self.env['product.image'].invalidate_model()
+            self.assertListEqual(self._get_product_image_data(), [i1, i2, i4, i3, i5, i6])
+
+    def test_resequence_image_last(self):
+        self._create_product_images()
+        with MockRequest(self.product.env, website=self.website):
             images = self.product._get_images()
-            self.assertEqual(images[2].image_1920, data_target)
-            self.assertEqual(images[3].image_1920, data_source)
-            # Test moving one to the last position
-            data_source = images[2].image_1920
-            data_target = images[-1].image_1920
+            i1, i2, i3, i4, i5, i6 = self._get_product_image_data()
             self.WebsiteSaleController.resequence_product_image(
                 images[2]._name,
                 images[2].id,
                 'last',
             )
+            self.env['product.image'].invalidate_model()
+            self.assertListEqual(self._get_product_image_data(), [i1, i2, i4, i5, i6, i3])
+
+    def test_resequence_image_first_to_last(self):
+        """
+        Moving an image from first to last position is an edge case.
+        This test ensures it won't be accidentally broken.
+        """
+        self._create_product_images()
+        with MockRequest(self.product.env, website=self.website):
+            images = self.product._get_images()
+            i1, i2, i3, i4, i5, i6 = self._get_product_image_data()
+            self.WebsiteSaleController.resequence_product_image(
+                images[0]._name,
+                images[0].id,
+                'last',
+            )
+            self.env['product.image'].invalidate_model()
+            self.assertListEqual(self._get_product_image_data(), [i2, i3, i4, i5, i6, i1])
+            self.assertEqual(self.product.image_1920, i2)
+
+    def test_resequence_video(self):
```

Comment:
```suggestion
    def test_resequence_video_left(self):
```

Reply:
Done (FYI the idea here was to have a test case for moving a video, not for a specific move, which is why I omitted the move type).